js/ui/html_editor/utils/image_uploader_helper.js

```diff
@@ -0,0 +1,457 @@
+import $ from '../../../core/renderer';
+import localizationMessage from '../../../localization/message';
+import { map } from '../../../core/utils/iterator';
+import devices from '../../../core/devices';
+const isMobile = devices.current().deviceType === 'phone';
+
+const DIALOG_IMAGE_CAPTION = 'dxHtmlEditor-dialogImageCaption';
+const DIALOG_IMAGE_FIELD_URL = 'dxHtmlEditor-dialogImageUrlField';
+const DIALOG_IMAGE_FIELD_ALT = 'dxHtmlEditor-dialogImageAltField';
+const DIALOG_IMAGE_FIELD_WIDTH = 'dxHtmlEditor-dialogImageWidthField';
+const DIALOG_IMAGE_FIELD_HEIGHT = 'dxHtmlEditor-dialogImageHeightField';
+const DIALOG_IMAGE_POPUP_CLASS = 'dx-htmleditor-add-image-popup';
+const DIALOG_IMAGE_POPUP_WITH_TABS_CLASS = 'dx-htmleditor-add-image-popup-with-tabs';
+const DIALOG_IMAGE_FIX_RATIO_CONTAINER = 'dx-fix-ratio-container';
+const FORM_DIALOG_CLASS = 'dx-formdialog';
+
+const USER_ACTION = 'user';
+const SILENT_ACTION = 'silent';
+
+import ButtonGroup from '../../button_group';
+import FileUploader from '../../file_uploader';
+import TextBox from '../../text_box';
+
+export class ImageUploader {
+    constructor(module, config) {
+        this.module = module;
+        this.config = config ?? {};
+
+        this.tabPanelIndex = 0;
+    }
+
+    render() {
+        this.formData = this.getFormData();
+        this.tabs = this.createTabs(this.formData);
+        const formConfig = this.getFormConfig();
+
+        this.modifyDialogPopupOptions();
+
+        this.module.editorInstance.showFormDialog(formConfig)
+            .done((formData, event) => {
+                this.tabs[this.getActiveTabIndex()].strategy.pasteImage(formData, event);
+            })
+            .always(() => {
+                this.cancelDialogPopupOptions();
+                this.module.quill.focus();
+            });
+    }
+
+    getActiveTabIndex() {
+        return this.tabPanelIndex;
+    }
+
+    getFormData() {
+        return this.getUpdateDialogFormData(this.module.quill.getFormat());
+    }
+
+    getUpdateDialogFormData(formData) {
+        const resultFormData = formData;
```

Comment:
```suggestion
        const { imageSrc, src, ...props } = formData;
        return { 
            src: imageSrc ?? src,
            ...props
        };
```

Or we really need to mutate the parameter?